sort.php

```diff
@@ -13,19 +13,29 @@ function checkForNextLine($arr)
     return $arr;
 }
 
-function bubbleSort($sort_array)
+function bubbleSort($sort_array, $options = false)
 {
     $array_indexes = count($sort_array);
     $sorted = false;
 
     for ($i = 0; $i < $array_indexes; $i++) {
         for ($j = 0; $j < $array_indexes - $i - 1; $j++) {
-            if ($sort_array[$j + 1] < $sort_array[$j]) {
-                $tmp = $sort_array[$j + 1];
-                $sort_array[$j + 1] = $sort_array[$j];
-                $sort_array[$j] = $tmp;
-
-                $sorted = true;
+            if ($options) {
+                if ((int)$sort_array[$j + 1] < (int)$sort_array[$j]) {
+                    $tmp = $sort_array[$j + 1];
+                    $sort_array[$j + 1] = $sort_array[$j];
+                    $sort_array[$j] = $tmp;
+
+                    $sorted = true;
+                }
+            } else {
+                if ($sort_array[$j + 1] < $sort_array[$j]) {
+                    $tmp = $sort_array[$j + 1];
+                    $sort_array[$j + 1] = $sort_array[$j];
+                    $sort_array[$j] = $tmp;
+
+                    $sorted = true;
+                }
```

Comment:
The only difference is the int conversion. The other stuff is duplicate.

Please replace the options parameter with a comparator parameter – a function which takes `$sort_array[$j + 1]` and `$sort_array[$j]` and returns whether the one is less than the other, so you don't need the if/else, but just call the function.

Depending on -n pass as comparator a function which casts to int or not.